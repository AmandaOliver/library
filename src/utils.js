import config from './configuration'
import React from 'react'

export const sortBooks = (property, books) => {
  if (property === 'id') {
    return books.sort((bookA, bookB) => bookA.id - bookB.id)
  }
  return books.sort((bookA, bookB) => bookA.bookData[property] < bookB.bookData[property] ? -1 : 1)
}

export const getBookPropertiesDropDown = (conf=config) => Object.keys(conf.bookProperties).map(property =>
  <option key={property} value={property} label={conf.bookProperties[property].label} />)

export const getPropertyValues = (property, conf=config) =>  conf.bookProperties[property].values
