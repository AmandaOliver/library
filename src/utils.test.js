import { sortBooks, getBookPropertiesDropDown, getPropertyValues } from './utils'
describe('utils: sortBooks', () => {
  it('sorts the books correctly by id', () => {
    const books = [
      {id: 2},
      {id: 1},
      {id: 0},
    ]
    expect(sortBooks('id', books)).toEqual([{id: 0}, {id:1}, {id:2}])
  })
  it('sorts the books correctly by property', () => {
    const books = [
      {bookData: { name: 'B'}},
      {bookData: { name: 'A'}},
      {bookData:{name: '1'}}
    ]
    const expectedBooks = [
      {bookData: { name: '1'}},
      {bookData: { name: 'A'}},
      {bookData:{name: 'B'}}
    ]
    expect(sortBooks('name', books)).toEqual(expectedBooks)
  })
})

describe('utils: bookPropertiesDropDown', () => {
  it('returns the options of the property dropdown populated with the labels', () => {
    const config = { bookProperties: { genre: { label: 'label1', name: 'label2' } } }
    expect(getBookPropertiesDropDown(config)).toMatchSnapshot()
  })
})

describe('utils: getPropertyValues', () => {
  it('returns the values for the book properties', () => {
    const config = { bookProperties: { genre: { values: ['genre1', 'genre2'] } } }
    expect(getPropertyValues('genre', config)).toEqual(['genre1', 'genre2'])
  })
})
