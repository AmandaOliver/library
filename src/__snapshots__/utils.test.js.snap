// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils: bookPropertiesDropDown returns the options of the property dropdown populated with the labels 1`] = `
Array [
  <option
    label="label1"
    value="genre"
  />,
]
`;
